nion/swift/ImageCanvasItem.py

```diff
@@ -255,34 +263,78 @@ def _repaint(self, drawing_context: DrawingContext.DrawingContext, canvas_bounds
         scale_marker_height = 6
         scale_marker_font = "normal 14px serif"
         get_font_metrics_fn = self.__get_font_metrics_fn
-        if dimensional_calibration:  # display scale marker?
+
+        if dimensional_calibration and screen_pixel_per_image_pixel and screen_pixel_per_image_pixel > 0.0:
+            scale_marker_image_width = scale_marker_width / screen_pixel_per_image_pixel
+            calibrated_scale_marker_width = Geometry.make_pretty2(scale_marker_image_width * dimensional_calibration.scale, True)
+            scale_marker_image_width = calibrated_scale_marker_width / dimensional_calibration.scale
+            scale_marker_calculated_width = scale_marker_image_width * screen_pixel_per_image_pixel
+
+
+            text1 = dimensional_calibration.convert_to_calibrated_size_str(scale_marker_image_width)
+            fm1 = get_font_metrics_fn(scale_marker_font, text1)
+            text_height = float(fm1.height-8)
+            if self.__include_info_text:
+                text2 = info_text
+                fm2 = get_font_metrics_fn(scale_marker_font, text2)
+                text_width = max(fm1.width, fm2.width)
+            else:
+                text2 = ""
+                text_height = text_height/2
+                text_width = fm1.width
+            padding = 3
+
+            content_width = max(scale_marker_calculated_width, text_width)
+            # Set drawing origin
+            if self.__scale_marker_position == "draw_bottom_right":
```

Comment:
Can you use the same constant style as the line plot canvas item legend? i.e. `bottom-right`?

As a style note: I try to use dashes when the constant is a string; and underscores when it is code so that it is less likely that someone will confuse the two.